tutorials/s4sdk-resilience/s4sdk-resilience.md

```diff
@@ -37,100 +37,132 @@ In order to avoid such scenarios, we need to equip applications with the ability
+
+At the same time we will also separate the VDM call itself into another class for better readability and easier maintenance in future tutorials. Note that starting with version 3 of the SAP Cloud SDK, a separate class is no longer required to implement resilience. We could have also added resilience directly to the existing `BusinessPartnerServlet` class.
+
+So first we will create the following class:
 
 `./application/src/main/java/com/sap/cloud/sdk/tutorial/GetBusinessPartnersCommand.java`
 
-```
+```java
 package com.sap.cloud.sdk.tutorial;
 
-import org.slf4j.Logger;
-
+import java.time.Duration;
 import java.util.Collections;
 import java.util.List;
 
-import com.sap.cloud.sdk.cloudplatform.logging.CloudLoggerFactory;
-
-import com.sap.cloud.sdk.s4hana.connectivity.ErpCommand;
-import com.sap.cloud.sdk.s4hana.datamodel.odata.helper.Order;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DestinationAccessor;
+import com.sap.cloud.sdk.cloudplatform.resilience.ResilienceConfiguration;
+import com.sap.cloud.sdk.cloudplatform.resilience.ResilienceDecorator;
+import com.sap.cloud.sdk.cloudplatform.resilience.ResilienceIsolationMode;
+import com.sap.cloud.sdk.cloudplatform.resilience.ResilienceRuntimeException;
+import com.sap.cloud.sdk.datamodel.odata.helper.Order;
+import com.sap.cloud.sdk.odatav2.connectivity.ODataException;
+
+import com.sap.cloud.sdk.s4hana.connectivity.DefaultErpHttpDestination;
+import com.sap.cloud.sdk.s4hana.connectivity.ErpHttpDestination;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.AddressEmailAddress;
 import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartner;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartnerAddress;
+import com.sap.cloud.sdk.s4hana.datamodel.odata.services.BusinessPartnerService;
 import com.sap.cloud.sdk.s4hana.datamodel.odata.services.DefaultBusinessPartnerService;
 
-public class GetBusinessPartnersCommand extends ErpCommand<List<BusinessPartner>> {
-    private static final Logger logger = CloudLoggerFactory.getLogger(GetBusinessPartnersCommand.class);
+public class GetBusinessPartnersCommand {
 
     private static final String CATEGORY_PERSON = "1";
+    private final ErpHttpDestination destination;
+
+    private final BusinessPartnerService businessPartnerService;
+    private final ResilienceConfiguration myResilienceConfig;
 
-    protected GetBusinessPartnersCommand() {
-        super(GetBusinessPartnersCommand.class);
+    public GetBusinessPartnersCommand(String destinationName) {
```

Comment:
Our best practice recommendation is to inject the destination itself, isn't it? For easier testability etc.

Reply:
The Command fetches the destination when instantiated by name. Fetching the destination once when the servlet is instantiated would not be desirable. Or do you mean something different with injecting?

Reply:
updated